<!-- run alongside: roslaunch nicol_rh8d dataset_cameras.launch marker_fps:=8 enable_depth:=false -->
<launch>

	<arg name="checkpoint_path" default="rh8d_all/validation/final_best_model_epoch=199_train_loss=0.018.ckpt" doc="Relative to MLP_CHKPT_PTH"/>
	<arg name="scalers_path" default="rh8d_all" doc="Relative to MLP_SCLRS_PTH"/>
	<arg name="markers_camera_name" default="markers_camera" /> <!-- set params as in dataset_cameras.launch -->

  	<arg name="attached"      default="false"/>
	<arg name="vis" default="true" />
	<arg name="save_imgs" default="true" />
	<arg name="f_ctrl" default="30" /> 
	<arg name="debug" default="false" /> 
	<arg name="test" default="false" /> 
	<arg name="use_tf" default="false" /> 
	<arg name="rh8d_port" default="/dev/ttyUSB0" />
	<arg name="rh8d_baud" default="1000000" />
	<arg name="gyccel_port" default="/dev/ttyACM0" />
	<arg name="gyccel_baud" default="115200" />

	<!-- demo script -->
	<node pkg="nicol_rh8d" name="dataset_collector" type="detect.py" output="screen" required="true">
		<param name="markers_camera_name" value="$(arg markers_camera_name)/color" />
		<param name="vis" value="$(arg vis)" />
		<param name="f_ctrl" value="$(arg f_ctrl)" />
		<param name="rh8d_port" value="$(arg rh8d_port)" />
		<param name="rh8d_baud" value="$(arg rh8d_baud)" />
		<param name="gyccel_port" value="$(arg gyccel_port)" />
		<param name="gyccel_baud" value="$(arg gyccel_baud)" />
		<param name="debug" value="$(arg debug)" /> 
		<param name="test" value="$(arg test)" /> 
		<param name="use_tf"     value="$(arg use_tf)" />
		<param name="attached"     value="$(arg attached)" />
		<param name="save_imgs"     value="$(arg save_imgs)" />
		<param name="checkpoint_path" value="$(arg checkpoint_path)" />
		<param name="scalers_path" value="$(arg scalers_path)" />
		<param name="demo" value="true" />
	</node>

</launch>
